### ISSUE TITLE
[4.0.0] indent of chaining methods

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#8709]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.0
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

```
{
    "extends": ["mysticatea", "mysticatea/node"],
    "rules": {
        "indent": ["error", 4, {"SwitchCase": 1, "flatTernaryExpressions": true}]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

- [no-hide-core-modules.js#L59-L64](https://github.com/mysticatea/eslint-plugin-node/blob/master/lib/rules/no-hide-core-modules.js#L59-L64)

```js
            const targets =
                [].concat(
                    getRequireTargets(context, true),
                    getImportExportTargets(context, node, true)
                )
                .filter(t => CORE_MODULES.has(t.moduleName))
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
C:\Users\starc\Documents\GitHub\eslint-plugin-node\lib\rules\no-hide-core-modules.js
  64:1  error  Expected indentation of 20 spaces but found 16  indent
```

**Comment by not-an-aardvark** [comment]
Hmm, I think this is the intended behavior -- the rule indents `MemberExpression` properties by 1 from the object.

**Comment by mysticatea** [comment]
Hmm, I see. I agreed you are right.

**Comment by jonathanGB** [comment]
Is it possible to add a "first" value to `MemberExpression`, so we can have the previous behaviour of vertically chaining to the same indentation as the first member (just like "first" for `FunctionExpression`).

**Comment by not-an-aardvark** [comment]
The previous behavior was to ignore this indentation entirely. You can achieve that with `MemberExpression: "off"`.

**Comment by jonathanGB** [comment]
That's what I've fell back to, but it seems to me like it's the logical next step, no?

**Comment by not-an-aardvark** [comment]
Maybe. The problem is that people have a wide variety of different indentation preferences, and the `indent` rule can't realistically support all of them. Instead, the rule has a limited set of options and allows you to turn off checks like `MemberExpression`. The idea is that you can use the rule even if you disagree with a particular part of what it enforces, and potentially use a custom rule to enforce your own preferences for `MemberExpression` indentation.

This isn't to say that we're never accepting any more options -- feel free to create a new issue if you'd like to propose a new one. But there is a possibility that it won't be accepted, simply because we can't support everything.

**Comment by jonathanGB** [comment]
Yeah I understand, you favour a "plug n' play" approach. Thanks for the explanation!

**Comment by G-Rath** [comment]
@not-an-aardvark when you say the previous behavior, `MemberExpression` wasn't added in 4.0.0 - we have code that was passing under eslint 3.19.0, but is now failing in 4.0.0, because of the changes made to `MemberExpression`.

From reading up on the indent rule, given what you've said and the existing options available for the indent rule, I think that `MemberExpression` is a prime candidate to have support for a `first` value.

This value is support for a number of other options, but not for MemberExpression - A better name such as `FirstWhenNewLine` might fit better, but I think it's a pretty valid indent rule to have. 

I'm really reluctant to just turn off an entire rule like that, since it can be very easy for new code to slip by undetected, that we previously even considered or realise that eslint would have caught.
